### PR TITLE
xplr: init at 0.5.4

### PR DESCRIPTION
**Pull request by sayanarijit**
maintainers: add sayanarijit
maintainers: add suryasr007

###### Motivation for this change
To add [xplr](https://github.com/sayanarijit/xplr) to nixpkgs. The changes were taken from [suryasr007/xplr-nix](https://github.com/suryasr007/nixpkgs/tree/xplr-nix) and has been reworked.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sayanarijit** [comment]
cc @suryasr007

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 119976` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>xplr</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -31057,4 +31057,6 @@ in
   lc3tools = callPackage ../development/tools/lc3tools {};
 
   zktree = callPackage ../applications/misc/zktree {};
+
+  xplr = callPackage ../applications/misc/xplr {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:31061]: Please sort this into a fitting category and sort there alphabetically.

File: pkgs/applications/misc/xplr/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  version = "v0.5.3";
+
+  src = fetchFromGitHub {
+    owner = "sayanarijit";
+    repo = name;
+    rev = version;
+    sha256 = "1l15agl20l2x80rswbg555z4gwi9v8v45vld09zd48cn3ixhj67d";
+  };
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/xplr/default.nix:14]: Why did you disable tests?

File: pkgs/applications/misc/xplr/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{ lib, rustPlatform, fetchFromGitHub }:
+
+rustPlatform.buildRustPackage rec {
+  name = "xplr";
+  version = "v0.5.3";
+
+  src = fetchFromGitHub {
+    owner = "sayanarijit";
+    repo = name;
+    rev = version;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/xplr/default.nix:10]:
```suggestion
  version = "0.5.3";

  src = fetchFromGitHub {
    owner = "sayanarijit";
    repo = name;
    rev = "v${version}";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (sayanarijit) [reply pkgs/top-level/all-packages.nix]: I think it belongs in the same category as `ranger`, `nnn`, `vifm` etc. i.e. `applications/misc`. Sorted.

File: pkgs/applications/misc/xplr/default.nix (hunk 2 above)
Comment (sayanarijit) [reply pkgs/applications/misc/xplr/default.nix]: Fixed.

File: pkgs/applications/misc/xplr/default.nix (hunk 3 above)
Comment (sayanarijit) [reply pkgs/applications/misc/xplr/default.nix:10]: Fixed.

**Review by SuperSandro2000 — APPROVED** [review]

**Review by legendofmiracles — APPROVED** [review]
Also gave a bit of feedback in the unofficial discord before they opened a PR
looks good now to me too

**Pull request merged**